internal/sets/mapset.go

```diff
@@ -0,0 +1,153 @@
+package sets
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+)
+
+// Similar to the concept of a [legal peppercorn](https://en.wikipedia.org/wiki/Peppercorn_(law)), this instance of
+// nothingness is required in order to transact with Go's map[T]struct{} idiom.
+var peppercorn = struct{}{}
+
+// MapSet is a struct that adds some convenience to the otherwise cumbersome map[T]struct{} idiom used in Go to
+// implement sets of comparable types.
+type MapSet[T comparable] struct {
+	m map[T]struct{}
+}
+
+// NewMapSet returns a MapSet ready for use. Optionally, a desired size for the MapSet can be passed as an argument,
+// as in the argument to make() for a map type.
+func NewMapSet[T comparable](args ...int) MapSet[T] {
```

Comment:
Make this could be `MakeMapSet` since it is kind of like `make(map[],n)`

Reply:
or just `Make` if we call the package `mapset`

Reply:
Groovy, if you feel like that's a better name I'm all for it.